### PR TITLE
Semihalf updates

### PR DESCRIPTION
**Pull request by semihalf-wojtas-marcin**
Hi,

Please check Semihalf Team updates from the 2022Q2.

Best regards,
Marcin

### CODE REVIEW

**Review by pauamma — CHANGES_REQUESTED** [review]

File: 2022q2/ena.adoc [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  * Extension of the timer service lifetime in order to be able to detect more
+    hardware failures,
+  * Fix logic for verifying the Tx request ID,
+  * Fix IPv6 L4 checksum offload handling for the Tx,
+  * Add NUMA awareness to the driver.
+* Internal review of the upcoming ENA driver release (v2.6.0), including:
+  * Further reset handling improvements,
+  * Code cleanup and style fixes,
+  * Logging improvements,
+  * Fix for the retrieval of the ENI metrics.
```
[the remark this answers is not in the document]
Comment (semihalf-wojtas-marcin) [reply 2022q2/ena.adoc]: Done.

File: 2022q2/ena.adoc [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+Contact: Dawid Gorecki <dgr@semihalf.com> +
+Contact: Marcin Wojtas <mw@FreeBSD.org>
+
+ENA (Elastic Network Adapter) is the smart NIC available in the virtualized environment of Amazon Web Services (AWS).
+The ENA driver supports multiple transmit and receive queues and can handle up to 100 Gb/s of network traffic, depending on the instance type on which it is used.
+
+Completed since the last update:
+
+* Upstream of the ENA driver v2.5.0, which was including:
+  * Improvement for the reset routine handling,
```
[the remark this answers is not in the document]
Comment (semihalf-wojtas-marcin) [reply 2022q2/ena.adoc]: Done.

File: 2022q2/ena.adoc [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+Contact: Michal Krawczyk <mk@semihalf.com> +
+Contact: Dawid Gorecki <dgr@semihalf.com> +
+Contact: Marcin Wojtas <mw@FreeBSD.org>
+
+ENA (Elastic Network Adapter) is the smart NIC available in the virtualized environment of Amazon Web Services (AWS).
+The ENA driver supports multiple transmit and receive queues and can handle up to 100 Gb/s of network traffic, depending on the instance type on which it is used.
+
+Completed since the last update:
+
+* Upstream of the ENA driver v2.5.0, which was including:
```
[the remark this answers is not in the document]
Comment (semihalf-wojtas-marcin) [reply 2022q2/ena.adoc]: Done.

File: 2022q2/shp_rand.adoc [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+Contact: Kornel Duleba <mindal@semihalf.com> +
+Contact: Marcin Wojtas <mw@FreeBSD.org>
+
+Shared page is an R/X page that is mapped into each process by the image activator.
+It stores the signal trampoline, as well as other metadata e.g. information needed
+to implement user space timecounters.
+Previously it was mapped at the top of the process virtual address space.
+With the described changes its address will be randomized.
+The feature planned to be turned on by default for 64bit binaries, across all architectures.
```
[the remark this answers is not in the document]
Comment (semihalf-wojtas-marcin) [reply 2022q2/shp_rand.adoc]: Done.

**Review by pauamma — APPROVED** [review]

**Pull request merged**